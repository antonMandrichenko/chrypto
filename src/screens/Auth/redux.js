import { userActions } from "../../store/actions/auth.actions";
import { fetchActions } from "../../store/actions/fetch.actions";

export const mapStateToProps = state => {
  return {
    isLoading: state.fetch.isLoading,
    errorLoading: state.fetch.error
  };
};

export const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    loginSuccess: data =>
      dispatch(userActions.loginSuccess(ownProps, dispatch, data)),
    loginFailure: data =>
      dispatch(userActions.loginFailure(ownProps, dispatch, data)),
    registrationSuccess: data =>
      dispatch(userActions.registrationSuccess(ownProps, data)),
    registrationFailure: err => dispatch(userActions.registrationFailure(err)),
    fetchBegin: () => dispatch(fetchActions.fetchBegin()),
    fetchSuccess: () => dispatch(fetchActions.fetchSuccess()),
    fetchFailure: err => dispatch(fetchActions.fetchFailure(err)),
    noResponse: () => dispatch(fetchActions.noResponse()),
    noFetch: () => dispatch(fetchActions.noFetch())
  };
};
