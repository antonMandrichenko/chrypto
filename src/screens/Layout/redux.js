// import { userActions } from "../../store/actions/auth.actions";

export const mapStateToProps = state => {
  return {
    id: state.registration.id || state.authentication.id,
  };
};
