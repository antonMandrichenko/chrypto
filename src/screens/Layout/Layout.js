import React from "react";
// import PropTypes from "prop-types";
import { withRouter, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { compose } from "redux";
import { mapStateToProps } from "./redux";
import Header from "./Header";
import RollDice from "./RollDice";
// import gql from "graphql-tag";
// import { useQuery } from "@apollo/react-hooks";

// const GET_ME = gql`
//   {
//     getMe {
//       id
//       affiliateCode
//       email
//       username
//     }
//   }
// `;

const propTypes = {};

function Layout(props) {
  const { id } = props;
  // const [userId, setUserId] = useState("");
  // const idUser = localStorage.getItem("cryptoId");
  // const username = localStorage.getItem("cryptoUsername");
  // const affiliateCode = localStorage.getItem("cryptoAffiliateCode");

  // let { loading, error, data } = useQuery(GET_ME);
  // if (data) {
  //   setUserId(data.id);
  // }

  // if (loading) return null;

  // if (!userId || error) {
  //   return <Redirect to="/auth" />
  // }
  // if (id) {
  //   return <div>`You are register in crypto. Yor're id is ${id}`</div>;
  // }
  // return id ? <div>You are in Crypto</div> : <Redirect to="/auth" />
  return (
    <React.Fragment>
      <Header />
      <RollDice />
    </React.Fragment>
  );
}

Layout.propTypes = propTypes;

export default compose(
  withRouter,
  connect(mapStateToProps)
)(Layout);
