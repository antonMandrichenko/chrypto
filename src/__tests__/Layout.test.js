import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Layout from "../screens/Layout";
import { createStore } from "redux";
import { Provider } from "react-redux";
import rootReducer from "../store/reducers";
import { MockedProvider } from "@apollo/react-testing";
import { MemoryRouter } from "react-router-dom";

const initialState = {};

function renderWithReduxApollo(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(
      <MockedProvider addTypename={false}>
        <Provider store={store}>{ui}</Provider>
      </MockedProvider>
    ),
    store
  };
}

it("loads and displays Layout", async () => {
  const { asFragment } = renderWithReduxApollo(
    <MemoryRouter>
      <Layout name="Buck" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});