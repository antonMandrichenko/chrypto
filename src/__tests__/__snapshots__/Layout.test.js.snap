// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays Layout 1`] = `<DocumentFragment />`;
